app/containers/ItunesContainer/index.js

```diff
@@ -4,46 +4,121 @@
  *
  */
 
-import React, { memo } from 'react';
+import React, { memo, useEffect, useState } from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import { injectIntl } from 'react-intl';
 import { createStructuredSelector } from 'reselect';
 import { compose } from 'redux';
 import { injectSaga } from 'redux-injectors';
-import { Input } from 'antd';
-import makeSelectItunesContainer from './selectors';
+import { Input, Skeleton } from 'antd';
+import makeSelectItunesContainer, { selectItuneName, selectItunesData, selectItunesError } from './selectors';
 import saga from './saga';
 import { itunesContainerCreators } from './reducer';
-import { debounce, isEmpty } from 'lodash';
+import { debounce, get, isEmpty } from 'lodash';
+import styled from 'styled-components';
+import ItunesCard from '@app/components/ItunesCard/index';
 
 const { Search } = Input;
 
-export function ItunesContainer({ dispatchItunesData, dispatchClearItunesData }) {
+const FlexContainer = styled.div`
+  display: grid;
+  grid-template-columns: repeat(3, 1fr);
+`;
+
+const Container = styled.div`
+  && {
+    max-width: 75rem;
+    width: 100%;
+    margin: 0 auto;
+    padding: 1rem;
+  }
+`;
+
+export function ItunesContainer({ dispatchItunesData, dispatchClearItunesData, itunesData, itunesError, ituneName }) {
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    const loaded = get(itunesData, 'items', null) || itunesError;
+    if (loaded) {
+      setLoading(false);
+    }
+  }, [itunesData]);
+
+  useEffect(() => {
+    if (ituneName && !itunesData?.results?.length) {
+      dispatchItunesData(ituneName);
+      setLoading(true);
+    }
+  }, []);
+
   const handleOnChange = (ituneName) => {
     if (!isEmpty(ituneName)) {
       dispatchItunesData(ituneName);
+      setLoading(true);
     } else {
       dispatchClearItunesData();
     }
   };
 
+  const [currTrack, setCurrTrack] = useState();
+
+  const onClickAction = (e, audioRef) => {
+    if (!isEmpty(currTrack)) {
+      if (currTrack === audioRef?.current) {
```

Comment:
What will be the value of `audioRef?.current` ?

Reply:
audioRef?.current? will be the reference of the audio which is playing.